arangod/VocBase/Methods/Indexes.cpp

```diff
@@ -170,7 +176,14 @@ arangodb::Result Indexes::getAll(LogicalCollection const* collection,
       idx->toVelocyPack(tmp, flags);
     }
     tmp.close();
-    trx.finish(res);
+
+    if (!inputTrx) {
+      Result res;
+      trx->finish(res);
```

Comment:
I still think that `res = trx->finish(res)` is more appropriate, as finish can be called with an ok() Result, but itself can fail. Then it would return an error, which will be ignored here.

Reply:
Sorry, that was an accident, will revert.